src/casp/cell_data_manager/sql/mako_helpers.py

```diff
@@ -0,0 +1,192 @@
+    """
+    return f"'{v}'"
+
+
+def _bool_value_processor(v: bool) -> str:
+    """
+    Convert the provided boolean to a string and capitalize it, making it suitable for SQL syntax.
+
+    :param v: The boolean value to be processed.
+    :return: The processed boolean in string format and uppercase."""
+    return str(v).upper()
+
+
+def _normalize_column_names(column_names: t.List[str]) -> t.List[str]:
+    """
+    Convert all column names to lowercase
+
+    :param column_names: The column names to process.
+    :return: Set of normalized columns
+    """
+    return [column_name.lower() for column_name in column_names]
+
+
+def _remove_duplicates(column_names: t.List[str]) -> t.List[str]:
+    """
+    Remove duplicates from a list of column names. Case-sensitive, so it is recommended to normalize the column names
+    list before using it
+
+    :param column_names: The column names to process.
+    :return: List of unique column names
+    """
+    seen = set()
+    unique_items = []
+
+    for s in column_names:
+        if s not in seen:
+            seen.add(s)
+            unique_items.append(s)
+
+    return unique_items
+
+
+def _process_column_names(column_names: t.List[str]) -> t.List[str]:
+    """
+    Normalize column names, get rid of duplicates and validate the output list
+
+    :param column_names: List of columns to process
+    :return: Normalized and validated list of columns
+    :raises ValueError: If any column does not pass the validation rule
+    """
+    column_names_normalized = _normalize_column_names(column_names=column_names)
+    column_names_normalized_unique = _remove_duplicates(column_names_normalized)
+
+    for column_name in column_names_normalized_unique:
+        validate_column_name(column_name=column_name)
+
+    return column_names_normalized_unique
+
+
+def select(column_names: t.List[str]) -> str:
+    """
+    Construct a SQL SELECT body from the provided list of columns.
+
+    :param column_names: A list of column names to be included in the SELECT clause. If the list is empty, all columns
+        are selected.
+    :raises ValueError: If any column name contains more than one period or is an empty string.
+
+    :return: A string with comma-separated values of columns for subsequent use in an SQL query.
+
+    **Example**
+
+    Using in Mako template::
+        <%!
+            from casp.cell_data_manager.sql import mako_helpers
+        %>
+        ${mako_helpers.select(["c.CELL_TYPE", "i.DATASET_ID", "c.CELL_TYPE"])}
+        from `${project}.${dataset}.table_name`
+        ...
+
+    Results in::
+        select c.cell_type, i.dataset_id, c.cell_type
+        from `project-name-1.test_dataset.table_name`
+        ...
+    """
+    processed_columns = _process_column_names(column_names=column_names)
+    return f"select {', '.join(processed_columns)}" if processed_columns else "*"
+
+
+def where(filters: t.Optional[t.Dict[str, t.Any]]) -> str:
```

Comment:
How difficult would it be to implement `<...>__not_in` and `<...>__not_eq`?

Reply:
https://github.com/cellarium-ai/cellarium-cloud/issues/108

I'll work on this pretty soon.